academic_observatory_workflows/workflows/unpaywall_data_feed_telescope.py

```diff
@@ -0,0 +1,326 @@
+from datetime import datetime, time, timedelta
+from typing import Generator, List, Tuple, Union
+
+import pendulum
+from academic_observatory_workflows.config import schema_folder as default_schema_folder
+from academic_observatory_workflows.workflows.unpaywall_telescope import (
+    UnpaywallRelease,
+)
+from airflow.exceptions import AirflowException
+from croniter import croniter
+from dateutil.relativedelta import relativedelta
+from observatory.platform.utils.airflow_utils import (
+    AirflowVars,
+    get_airflow_connection_password,
+)
+from observatory.platform.utils.file_utils import find_replace_file, gunzip_files
+from observatory.platform.utils.http_download import download_file
+from observatory.platform.utils.url_utils import (
+    get_http_response_json,
+    get_observatory_http_header,
+)
+from observatory.platform.utils.workflow_utils import is_first_dag_run
+from observatory.platform.workflows.stream_telescope import (
+    StreamRelease,
+    StreamTelescope,
+)
+
+
+class UnpaywallDataFeedRelease(StreamRelease):
+    """Unpaywall Data Feed Release"""
+
+    AIRFLOW_CONNECTION = "unpaywall_data_feed"  # Contains API key
+
+    # These links are publicly listed on Unpaywall's website. See https://unpaywall.org/products/data-feed
+    SNAPSHOT_URL = "https://api.unpaywall.org/feed/snapshot"
+    CHANGEFILES_URL = "https://api.unpaywall.org/feed/changefiles"
+
+    def __init__(
+        self,
+        *,
+        dag_id: str,
+        start_date: pendulum.DateTime,
+        end_date: pendulum.DateTime,
+        first_release: bool,
+    ):
+        """Construct an UnpaywallDataFeedRelease instance
+
+        :param dag_id: the id of the DAG.
+        :param start_date: the start_date of the release.
+        :param end_date: the end_date of the release.
+        :param first_release: whether this is the first release that is processed for this DAG
+        """
+
+        super().__init__(
+            dag_id,
+            start_date,
+            end_date,
+            first_release,
+        )
+
+        self.http_header = get_observatory_http_header(package_name="academic_observatory_workflows")
+
+    @property
+    def api_key(self) -> str:
+        """The API key for accessing Unpaywall."""
+
+        return get_airflow_connection_password(UnpaywallDataFeedRelease.AIRFLOW_CONNECTION)
+
+    @property
+    def snapshot_url(self) -> str:
+        """Snapshot URL"""
+
+        return f"{UnpaywallDataFeedRelease.SNAPSHOT_URL}?api_key={self.api_key}"
+
+    @property
+    def data_feed_url(self) -> str:
+        """Data Feed URL"""
+
+        return f"{UnpaywallDataFeedRelease.CHANGEFILES_URL}?interval=day&api_key={self.api_key}"
+
+    def download(self):
+        """Download the release."""
+        # May need to refactor depending on how snapshot looks.
+        if self.first_release:
+            self.download_snapshot_()
+        else:
+            self.download_data_feed_()
+
+    def download_snapshot_(self):
```

Comment:
Private methods in Python should begin with an underscore, not end with an underscore, e.g. _download_snapshot.